src/ray/stats/stats.h

```diff
@@ -50,14 +50,20 @@ static inline void Init(
     RAY_LOG(INFO) << "Disabled stats.";
     return;
   }
+  RAY_LOG(DEBUG) << "Initialize stats";
 
   // Force to have a singleton exporter.
   static std::shared_ptr<MetricExporterClient> exporter;
-  // Default exporter is metrics agent exporter.
+  // Default exporter is metrics agent exporter and it will be register to stats
+  // if metric agent port is valid.
   if (exporter_to_use == nullptr) {
     std::shared_ptr<MetricExporterClient> stdout_exporter(new StdoutExporterClient());
-    exporter.reset(new MetricsAgentExporter(stdout_exporter, metrics_agent_port,
-                                            io_service, "127.0.0.1"));
+    if (metrics_agent_port > 0) {
+      exporter.reset(new MetricsAgentExporter(stdout_exporter, metrics_agent_port,
```

Comment:
Why does the `MetricsAgentExporter` need a reference to the `stdout_exporter`? Unintuitive to me.

Reply:
MetricExporterClient is decorator pattern that we talked in design doc https://docs.google.com/document/d/1ptWoY-14989mCTaz-inSO8XkoM_KrzT8bWwXXEv9Yzo/edit#.